Accordion.js

```diff
@@ -0,0 +1,474 @@
+/** @module deliteful/Accordion */
+define(["dcl/dcl",
+	"decor/sniff",
+	"requirejs-dplugins/Promise!",
+	"delite/register",
+	"delite/KeyNav",
+	"requirejs-dplugins/jquery!attributes/classes",
+	"delite/DisplayContainer",
+	"./ToggleButton",
+	"delite/theme!./Accordion/themes/{{theme}}/Accordion.css"
+], function (dcl, has, Promise, register, KeyNav, $, DisplayContainer, ToggleButton) {
+
+	function setVisibility(node, val) {
+		node.style.display = val ? "" : "none";
+	}
+
+	var animationEndEvent = (function () {
+		var animationEndEvents = {
+			"animation": "animationend", // > IE10, FF
+			"-webkit-animation": "webkitAnimationEnd",   // > chrome 1.0 , > Android 2.1 , > Safari 3.2
+			"-ms-animation": "MSAnimationEnd" // IE 10
+		};
+		// NOTE: returns null if event is not supported
+		var fakeElement = document.createElement("fakeElement");
+		for (var event in animationEndEvents) {
+			if (fakeElement.style[event] !== undefined) {
+				return animationEndEvents[event];
+			}
+		}
+		return null;
+	})();
+
+	function listenAnimationEndEvent(element) {
+		if (animationEndEvent) {
+			return new Promise(function (resolve) {
+				var handler = element.on(animationEndEvent, function () {
+					handler.remove();
+					resolve();
+				});
+			});
+		}
+	}
+
+	/**
+	 * A layout container that display a vertically stacked list of Panels whose titles are all visible, but only one
+	 * or at least one panel's content is visible at a time (depending on the singleOpen property value).
+	 *
+	 * Once the panels are in an accordion, they become collapsible Panels by replacing their headers by ToggleButtons.
+	 *
+	 * When a panel is open, it fills all the available space with its content.
+	 *
+	 * @example:
+	 * <d-accordion id="accordion" selectedChildId="panel1">
+	 *     <d-panel id="panel1">...</d-panel>
+	 *     <d-panel id="panel2">...</d-panel>
+	 *     <d-panel id="panel3">...</d-panel>
+	 * </d-accordion>
+	 * @class module:deliteful/Accordion
+	 * @augments module:delite/DisplayContainer
+	 */
+	var Accordion = dcl([DisplayContainer, KeyNav], /** @lends module:deliteful/Accordion# */ {
+
+		/**
+		 * The name of the CSS class of this widget.
+		 * @member {string}
+		 * @default "d-accordion"
+		 */
+		baseClass: "d-accordion",
+
+		/**
+		 * The id of the panel to be open at initialization.
+		 * If not specified, the default open panel is the first one.
+		 * @member {string}
+		 * @default ""
+		 */
+		selectedChildId: "",
+
+		/**
+		 * If true, only one panel is open at a time.
+		 * If false, several panels can be open at a time, but there's always at least one open.
+		 * @member {boolean}
+		 * @default true
+		 */
+		singleOpen: true,
```

Comment:
It might be more future proof to not use a boolean value. For example if one day we want to include a 3rd mode (like ability to close all panels) we will be stuck. For this reason, except in cases where I don't foresee a possible 3rd mode for the future, I usually tend to prefer a "mode" with various values. That said I don't have a very strong opinion.